File: app/src/main/java/com/example/todos/TaskAdapter.java
package com.example.todos;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;

import java.util.ArrayList;

public class TaskAdapter extends ArrayAdapter<Task> {
    private final Context mContext;
    private final ArrayList<Task> tasks;
    private final MainActivity mainActivity;

    public TaskAdapter(Context context, ArrayList<Task> tasks, MainActivity mainActivity) {
        super(context, 0, tasks);
        this.mContext = context;
        this.tasks = tasks;
        this.mainActivity = mainActivity;
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        // Pobierz dane dla tej pozycji
        Task currentTask = getItem(position);
        // Sprawdź, czy istniejący widok jest używany ponownie
        if (convertView == null) {
            convertView = LayoutInflater.from(mContext).inflate(R.layout.task_item, parent, false);
        }
        // Znajdź widoki do modyfikacji w widoku elementu
        CheckBox taskCheckBox = convertView.findViewById(R.id.taskCheckBox);
        TextView taskNameTextView = convertView.findViewById(R.id.taskNameTextView);
        ImageView taskIconImageView = convertView.findViewById(R.id.taskIconImageView);
        TextView taskDateTextView = convertView.findViewById(R.id.taskDateTextView);
        TextView taskTimeTextView = convertView.findViewById(R.id.taskTimeTextView);
        TextView taskNotesTextView = convertView.findViewById(R.id.taskNotesTextView);

        assert currentTask != null;
        taskNameTextView.setText(currentTask.getTaskName());
        taskCheckBox.setChecked(currentTask.isCompleted());

        if (currentTask.getIconResId() != -1) {
            taskIconImageView.setImageResource(currentTask.getIconResId());
            taskIconImageView.setVisibility(View.VISIBLE);
        } else {
            taskIconImageView.setVisibility(View.GONE);
        }
        //Ustaw date
        taskDateTextView.setText(currentTask.getDate());
        // Ustaw godzinę
        taskTimeTextView.setText(currentTask.getTime());
        // Ustaw notatki
        taskNotesTextView.setText(currentTask.getNotes());
        // Ustaw listener dla checkboxa
        taskCheckBox.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (isChecked) {
                tasks.remove(currentTask); // Usuń zadanie z listy
                notifyDataSetChanged();   // Informuj adapter o zmianie danych
                mainActivity.saveTasksToFile();
            }
        });
        return convertView;
    }
}

File: app/src/main/java/com/example/todos/TaskAdapter.kt
package com.example.todos

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.CheckBox
import android.widget.ImageView
import android.widget.TextView

class TaskAdapter(
    private val mContext: Context,
    private val tasks: MutableList<Task?>,
    private val mainActivity: MainActivity
) : ArrayAdapter<Task>(mContext, 0, tasks) {

    private class ViewHolder(
        val taskCheckBox: CheckBox,
        val taskNameTextView: TextView,
        val taskIconImageView: ImageView,
        val taskDateTextView: TextView,
        val taskTimeTextView: TextView,
        val taskNotesTextView: TextView
    )

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val currentTask = getItem(position)
        val viewHolder: ViewHolder
        val view: View
        if (convertView == null) {
            val inflater = LayoutInflater.from(mContext)
            view = inflater.inflate(R.layout.task_item, parent, false)
            viewHolder = ViewHolder(
                view.findViewById(R.id.taskCheckBox),
                view.findViewById(R.id.taskNameTextView),
                view.findViewById(R.id.taskIconImageView),
                view.findViewById(R.id.taskDateTextView),
                view.findViewById(R.id.taskTimeTextView),
                view.findViewById(R.id.taskNotesTextView)
            )
            view.tag = viewHolder
        } else {
            viewHolder = convertView.tag as ViewHolder
            view = convertView
        }
        currentTask?.let {
            with(viewHolder) {
                taskNameTextView.text = it.title
                taskCheckBox.isChecked = it.isCompleted

                if (it.iconResId != -1) {
                    taskIconImageView.setImageResource(it.iconResId)
                    taskIconImageView.visibility = View.VISIBLE
                } else {
                    taskIconImageView.visibility = View.GONE
                }
                taskDateTextView.text = it.date
                taskTimeTextView.text = it.time
                taskNotesTextView.text = it.notes
                taskCheckBox.setOnCheckedChangeListener { _, isChecked ->
                    if (isChecked) {
                        tasks.remove(it) // Usuń zadanie z listy
                        notifyDataSetChanged() // Informuj adapter o zmianie danych
                        mainActivity.saveTasksToFile()
                    }
                }
            }
        }
        return view
    }
}
